lib/state_machine/state_machine.cpp

```diff
@@ -0,0 +1,27 @@
+#include "state_machine.hpp"
+
+namespace hyped::state_machine {
+
+// TODOLater implement
+std::optional<Message> StateMachine::handleData()
+{
+  /*
+  check if a transition is needed with current data
+  then run transition to move to next state
+  return transition message or no transition
+  */
+  return std::nullopt;
+}
+
+// Transition to next state
+void StateMachine::handleMessage(const Message message)
+{
+  current_state = transitions.at({current_state, message});
+}
+
+void StateMachine::reset()
```

Comment:
I agree. In fact I don't like this function at all. You should get rid of it in order to avoid bugs. The user can construct a new state machine if that's what's required.